python/flatbuffers/table.py

```diff
@@ -53,6 +53,13 @@ def String(self, off):
         length = encode.Get(N.UOffsetTFlags.packer_type, self.Bytes, off)
         return bytes(self.Bytes[start:start+length])
 
+    def UnionString(self, off):
+        """String gets a string from data stored inside the flatbuffer."""
```

Comment:
This docstring should be updated, its not describing `UnionString`.

Do you think we should implement `String` with this method? They're pretty similar. Also, maybe this should be called `DirectString` or something since it makes sense outside of unions too.

Reply:
True. The simplest solutions are probably:

- Calling it `DirectString` or similar and having `String` call it after adjusting the offset
- Creating a method internal method `_DirectString` for both functions to use
- Having the offset adjustment be optional with a kwarg (defaulting to adjusting offset for backwards compatability)

However, these solutions do not feel good to me, as they make the usability of the API unintuitive. The user has no reason to think that strings coming out of unions should be treated differently when reading from the flatbuffer, which may cause confusion if they're not using a creator method to read it for them.

I can take a look at how the other supported languages deal with this, unless you have a specific solution in mind.

Reply:
> Creating a method internal method _DirectString for both functions to use

That seems fine to me if having a method called `UnionString` is critical for users. However, I'm not sure the typical user will call this method.

>  they make the usability of the API unintuitive. The user has no reason to think that strings coming out of unions should be treated differently when reading from the flatbuffer, which may cause confusion if they're not using a creator method to read it for them.

I wouldn't expect users to use `UnionString`, that should be hidden behind our generated code. Even your tests of this feature don't directly invoke `UnionString`.

Reply:
Fair enough. Sounds good to me, I'll make the suggested change